File: frontend/src/services/notificationService.ts
import { apiClient } from '@/config/apiClient';
import type { ApiResponse, PageResponse, TableParams } from '@/types';

export type NotificationType = 'INFO' | 'WARNING' | 'SUCCESS' | 'ERROR';

export interface NotificationResponse {
  id: number;
  title: string;
  content: string;
  type: NotificationType;
  isRead: boolean;
  createdAt: string;
  createdBy: string;
}

export interface NotificationFilterParams extends TableParams {
  isRead?: boolean;
}

export const notificationService = {
  getNotifications: (params?: NotificationFilterParams): Promise<ApiResponse<PageResponse<NotificationResponse>>> => {
    return apiClient.get('/notifications', { params });
  },

  getUnreadCount: (): Promise<ApiResponse<number>> => {
    return apiClient.get('/notifications/unread-count');
  },

  markAsRead: (id: number): Promise<ApiResponse<void>> => {
    return apiClient.put(`/notifications/${id}/read`);
  },

  markAllAsRead: (): Promise<ApiResponse<void>> => {
    return apiClient.put('/notifications/read-all');
  },
};

File: backend/src/main/java/com/scim/service/NotificationService.java
package com.scim.service;

import com.scim.dto.response.NotificationResponse;
import com.scim.dto.response.PageResponse;
import com.scim.entity.NotificationType;
import org.springframework.data.domain.Pageable;

public interface NotificationService {
    NotificationResponse create(Long userId, String title, String content, NotificationType type);
    PageResponse<NotificationResponse> getMyNotifications(Boolean isRead, Pageable pageable);
    long getMyUnreadCount();
    void markAsRead(Long id);
    void markAllAsRead();
}
